src/widgets/Notebook.cpp

```diff
@@ -1325,17 +1325,19 @@ SplitNotebook::SplitNotebook(Window *parent)
             // Note that selected tabs are always shown regardless of what the tab
             // filter returns, so no need to include `tab->isSelected()` in the
             // predicate. See Notebook::setTabVisibilityFilter.
-            switch (visibility)
-            {
-                case NotebookTabVisibility::LiveOnly:
-                    this->setTabVisibilityFilter([](const NotebookTab *tab) {
-                        return tab->isLive();
-                    });
-                    break;
-                case NotebookTabVisibility::AllTabs:
-                default:
-                    this->setTabVisibilityFilter(nullptr);
-                    break;
+            if ((visibility & (NotebookTabVisibility::LiveOnly | NotebookTabVisibility::UnreadOnly)) != 0) {
```

Comment:
You could make use of our `FlagsEnum` class to make the comparisons a little bit more readable.

Additionally, if possible, I'd prefer if the actual visibility-callbacks for these states stay as simple as possible. Meaning if someone has set the filter to "live only", we should be able to use the `{ return tab->isLive(); } ` callback
If someone has set the filter to "unread only" we should be able to use the `{ return tab->highlightState() != HighlightState::None; }` callback

a lot of yapping, but move as much of the logic out from the callback as possible, since it's run many times compared to the setting changing which is called less often